CRM/Contribute/Form/Contribution/Confirm.php

```diff
@@ -1929,6 +1934,18 @@ protected function processFormSubmission($contactID) {
     // fix currency ID
     $this->_params['currencyID'] = CRM_Core_Config::singleton()->defaultCurrency;
 
+    // CRM-18854
+    if (CRM_Utils_Array::value('adjust_recur_start_date', $this->_values)) {
+      $pledgeBlock = CRM_Pledge_BAO_PledgeBlock::getPledgeBlock($this->_id);
+      if (CRM_Utils_Array::value('start_date', $this->_params) || !CRM_Utils_Array::value('is_pledge_start_date_visible', $pledgeBlock)) {
+        $pledgeStartDate = CRM_Utils_Array::value('start_date', $this->_params, NULL);
+        $this->_params['receive_date'] = CRM_Pledge_BAO_Pledge::getPledgeStartDate($pledgeStartDate, $pledgeBlock);
+        $recurParams = CRM_Pledge_BAO_Pledge::buildRecurParams($this->_params);
+        $this->_params = array_merge($this->_params, $recurParams);
```

Comment:
The objective was to create a recur record for the pledge. So, the buildRecurParams generates the array needed for core to create the recur record by adding is_recur, frequency_interval etc.
Normally, it would be sufficient to just assign it to $this->_params, but there are instances in core that check $this->_values instead of $this->_params, eg. https://github.com/civicrm/civicrm-core/blob/master/CRM/Contribute/BAO/Contribution/Utils.php#L123

We do not know how many places such as these require the use of $this->_values rather than $this->_params, so even though it might seem like code done 'just in case', it really is quite necessary.